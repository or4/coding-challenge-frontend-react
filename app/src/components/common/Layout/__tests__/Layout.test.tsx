import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { Layout } from '..';

describe('Layout', () => {
    it('should correct render', () => {
        const tree = renderer.create(<Layout />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should apply children as component', () => {
        const Children = () => <div>There are some children</div>;
        const wrapper = shallow(<Layout>{<Children />}</Layout>);

        expect(wrapper.find(Children)).toHaveLength(1);
    });

    it('should apply children as jsx prop', () => {
        const children = <div className="test-children">There are some children</div>;
        const wrapper = shallow(<Layout>{children}</Layout>);

        expect(wrapper.find('.test-children')).toHaveLength(1);
    });
});
