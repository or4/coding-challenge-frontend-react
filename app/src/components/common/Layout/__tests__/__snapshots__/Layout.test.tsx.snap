// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should correct render 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c2 {
  width: 64px;
  height: 64px;
}

.c3 {
  padding-left: 15px;
}

.c4 {
  font-weight: 600;
}

.c5 {
  font-weight: 600;
}

.c0 {
  margin: 0 auto 15px;
}

@media (max-width:479px) {
  .c1 {
    height: 60px;
    margin-bottom: 15px;
  }
}

@media (min-width:480px) and (max-width:991px) {
  .c1 {
    height: 100px;
  }
}

@media (min-width:992px) {
  .c1 {
    height: 140px;
  }
}

@media (min-width:480px) {
  .c2 {
    padding: 0 18px;
  }
}

@media (max-width:479px) {
  .c4 {
    font-size: 18px;
  }
}

@media (min-width:480px) and (max-width:991px) {
  .c4 {
    font-size: 32px;
  }
}

@media (min-width:992px) {
  .c4 {
    font-size: 48px;
  }
}

@media (max-width:479px) {
  .c5 {
    font-size: 14px;
  }
}

@media (min-width:480px) and (max-width:991px) {
  .c5 {
    font-size: 16px;
  }
}

@media (min-width:992px) {
  .c5 {
    font-size: 32px;
  }
}

@media (max-width:479px) {
  .c0 {
    padding: 15px;
  }
}

@media (min-width:480px) and (max-width:991px) {
  .c0 {
    padding: 15px;
  }
}

@media (min-width:992px) {
  .c0 {
    max-width: 940px;
  }
}

<div
  className="c0"
  data-test-id="layout"
>
  <div
    className="c1"
  >
    <img
      className="c2"
      src="logo.png"
    />
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        Police Department of Berlin
      </div>
      <div
        className="c5"
      >
        Stolen bikes
      </div>
    </div>
  </div>
</div>
`;
